config/schemas/config.yaml

```
@@ -126,6 +340,702 @@ properties:
           - containerd
           - docker
     additionalProperties: false
+  clusterApi:
+    additionalProperties: false
+    properties:
+      clusters:
+        items:
+          type: string
+        type: array
+      enabled:
+        default: false
+        type: boolean
+      monitoring:
+        additionalProperties: false
+        properties:
+          enabled:
+            default: false
+            type: boolean
+        type: object
+    type: object
+  user:
+    additionalProperties:
+      $ref: '#/$defs/commonResource'
```

Would it be possible to make it ref directly to the common definition? Because with doc generators this becomes somewhat strange in that `additionalProperties` itself is reported as a property, although it should basically merge its definition with it instead.

---

`additionalProperties` applies to any property not described in `properties`, so this thing is meant as a short-cut to apply to anything new added under `.user` (in this case), so it validates as long as it has those common set of properties (enabled, affinity, resources...). This is partly because I haven't found everything that's possible to specify there.

Elsewhere `additionalProperties` is most often set to `false` to raise a "additional property <foo> is not allowed" error, so it is discovered.

```suggestion
    additionalProperties: false
```